.github/workflows/main.yaml

```diff
@@ -39,88 +39,28 @@ jobs:
           uv pip install ruff
         continue-on-error: false
 
-      - name: run ruff
-        run: | 
-          . .venv/bin/activate
-          ruff --output-format=github .
-        continue-on-error: false
-        
-
-  unit-tests:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.10"
-
-      - name: Install UV, venv
-        run: |
-          curl -LsSf https://astral.sh/uv/install.sh | sh
-          uv venv
-          echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
-
-      - name: Install Poetry
-        id: poetry
+      - name: Install Poetry and test dependencies
         run: |
           . .venv/bin/activate
           uv pip install poetry
-        continue-on-error: false
-
-      - name: Lock Poetry
-        run: |
-          . .venv/bin/activate
           poetry lock
-
-      - name: Install test dependencies
-        run: |
-          . .venv/bin/activate
           poetry install
-        continue-on-error: true
-
-      - name: Run unit tests
-        run: |
-          . .venv/bin/activate
-          poetry run pytest tests/unit
-        continue-on-error: false
-
-  integration-tests:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.10"
-
-      - name: Install UV, venv
-        run: |
-          curl -LsSf https://astral.sh/uv/install.sh | sh
-          uv venv
-          echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
-
-      - name: Install poetry
-        id: poetry
-        run: |
-          . .venv/bin/activate
-          uv pip install poetry 
         continue-on-error: false
 
-      - name: Lock Poetry
-        run: |
+      - name: Linting with Ruff
```

Comment:
I think this step should be moved up above the `Poetry and dependencies step`.

Since we are installing Ruff individually its very fast and can give feedback on job status before the longer running Poetry install (or the unit/integration tests).

Should we also halt the entire workflow if Ruff fails, before kicking off the heavier testing steps?

Reply:
sg! ty for the catch on the ordering.